### DISCUSSION TITLE
Set Up Vector for Effects

### DISCUSSION THREAD

**Discussion by Gamingtoday093** [SmartlyDressedGames/Unturned-3.x-Community#3687] [Ideas]
Effects currently use `Quaternion.LookRotation(Direction, Vector3.up);` to set their orientation, which limits the possible directions an effect can point since the upwards direction is always Vector3.up. I suggest adding the ability to specify the upwards direction to make an effect have any orientation.

https://github.com/Unturned-Datamining/Unturned-Datamining/blob/linux-preview/Assembly-CSharp/SDG.Unturned/EffectManager.cs#L1268

**Comment by GazziFX** [comment]
I think its better to convert it to euler angles

**Reply by Gamingtoday093** [reply]
No way worse. Most if not all the math to get the direction of an effect is done using Vectors and Quaternion. Being forced to convert it to euler angled would be a pain. No need to. I guess making it a Quaternion would sorta make sense but honestly Vectors are just the easiest to work with and they are already being used to determine the direction. Adding one more Vector3 to set the upwards direction should be an easy job now especially since the `TriggerEffectParameters` prevent issues with backwards compatability.

**Reply by GazziFX** [reply]
Euler angles is also just a Vector3 but instead of direction it gives a much reliable result and also Quaternion.LookRotation(Vector3.zero) gives warning.
Sending two Vector3 for direction and upwards is overhead

**Reply by GazziFX** [reply]
You also dealing with euler angles in Unity Editor why you need something else?
You can safely convert direction to angles but you can't get exact angles if you converted it to direction

**Reply by Gamingtoday093** [reply]
Its way easier to just specify the direction using Vectors instead of having the convert them to Euler Angles.
> You can safely convert direction to angles but you can't get exact angles if you converted it to direction

Wdym?

**Reply by GazziFX** [reply]
> Wdym?

You can't define rotation around direction axis

**Reply by Gamingtoday093** [reply]
Yes you can, you just have to specify what the Upwards direction is which is literally the basis of this Suggestion.

I agree that adding another Vector would result in overhead though. As long as the `TriggerEffectParameters` get helpers for all the ways youd want to orient an effect like Nelson proposed it will be fine.
